modules/shared/src/main/scala/org/tesselation/ext/crypto.scala

```diff
@@ -0,0 +1,30 @@
+package org.tesselation.ext
+
+import cats.MonadThrow
+import cats.syntax.either._
+
+import org.tesselation.crypto.hash.Hash
+import org.tesselation.crypto.{Hashable, Signed}
+import org.tesselation.kryo.KryoSerializer
+
+object crypto {
+  implicit class RefinedHashable[F[_]: KryoSerializer](anyRef: AnyRef) {
+
+    def hash: Either[Throwable, Hash] = Hashable.forKryo[F].hash(anyRef)
+  }
+
+  implicit class RefinedHashableF[F[_]: MonadThrow: KryoSerializer](anyRef: AnyRef) {
+
+    def hashF: F[Hash] = Hashable.forKryo[F].hash(anyRef).liftTo[F]
+  }
+
+  implicit class RefinedSigned[F[_]: KryoSerializer, A <: AnyRef](data: A) {
+
+    def sign(): Either[Throwable, Signed[A]] = Signed.forKryo[F, A](data)
+  }
+
+  implicit class RefinedSignedF[F[_]: KryoSerializer, A <: AnyRef](data: A) {
+
+    def signF[M[_]: MonadThrow](): M[Signed[A]] = Signed.forKryo[F, A](data).liftTo[M]
+  }
+}
```

Comment:
Nice, I like that you're adding syntax for core functionalities 👍

Reply:
Nice. Did you check if the possibility of mixup exists? So the situation where I expected that I'm fetching this hash, but I'm actually getting some scala default implementation of a hash?

Reply:
@TheMMaciek not yet

Reply:
but actually you would want `Hash` or `Signed` type classes. default implementation of `.hash` returns Int probably

Reply:
Yep, relying on the Hash and Signed should give us a lot of safety.